bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/handler/SunSynkInverterHandler.java

```diff
@@ -0,0 +1,478 @@
+                            1);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_2_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            2);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_3_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            3);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_4_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            4);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_5_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            5);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_6_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            6);
+                    break;
+                // Inverter control
+                case CHANNEL_INVERTER_CONTROL_TIMER:
+                    this.tempInverterChargeSettings.setPeakAndValley(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_ENERGY_PATTERN:
+                    this.tempInverterChargeSettings.setEnergyMode(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_WORK_MODE:
+                    this.tempInverterChargeSettings.setSysWorkMode(Integer.valueOf(command.toString()));
+                    break;
+            }
+
+            Optional<SunSynkAccountHandler> checkBridge = getSafeBridge();
+            if (!checkBridge.isPresent()) {
+                logger.debug("Failed to find associated SunSynk Bridge.");
+                return;
+            }
+            this.batterySettingsUpdated = true; // true = update battery settings from API at next interval
+        }
+    }
+
+    private void sendAPICommandToInverter(Settings inverterChargeSettings) {
+        logger.debug("Ok - will handle command for CHANNEL_BATTERY_INTERVAL_1_GRID_CHARGE");
+        SunSynkInverterConfig config = getThing().getConfiguration().as(SunSynkInverterConfig.class);
+        // Settings inverterChargeSettings = inverter.getBatteryChargeSettings();
+        String body = inverterChargeSettings.buildBody();
+        String token = inverterChargeSettings.getToken();
+        String response = inverter.sendCommandToSunSynk(body, token);
+
+        if ("Authentication Fail".equals(response)) { // try refreshing log in
```

Comment:
If the service returns 401 or 403 status code, it would be more reliable to use that as a check.

Reply:
Your suggestion was how the code worked when it was written using HttpsURLConnection, but I understood it was preferred to use org.openhab.core.io.net.http.HttpUtil which doesn't expose the underlying error code as far as I can tell, happy to be corrected, via an example.